src/main/java/org/folio/linked/data/mapper/dto/monograph/work/sub/ReferenceMapperUnit.java

```diff
@@ -28,7 +31,9 @@ public <P> P toDto(Resource source, P parentDto, Resource parentResource) {
   @Override
   public Resource toEntity(Object dto, Resource parentEntity) {
     var reference = (Reference) dto;
-    return fetchResource(reference, resourceRepository);
+    return fetchResource(reference, resourceRepository)
```

Comment:
Hi @AndreiBordak,
Here also we need to apply the same logic  you implemented in AgentMapper.
1. AgentMapper is used for "creator of work"
2. ReferenceMapper is used for all other kinds of references (like subject of work, genre/form, ......) 


Maybe we can move the `fetchResource` method from `BibframeUtils` to this class as a private method.

Reply:
done.